.github/workflows/build.yml

```diff
@@ -8,6 +8,7 @@ on:
 
 env:
    CARGO_TERM_COLOR: always
+   RUSTFLAGS:        -Dwarnings
```

Comment:
this is needed to fail on warnings, see [the clippy ci docs](https://doc.rust-lang.org/stable/clippy/continuous_integration/github_actions.html).